### ISSUE TITLE
[MU4 Issue] Trying to use soundfonts doesn't work.

### ISSUE THREAD

**Issue by DaforLynx** [musescore/MuseScore#15003]
**Describe the bug**
I have...a lot of soundfonts. When I try to load the Soundfonts dropdown menu in the mixer, the app hangs for a whole minute. Then attempting to mouse over the patches just hangs it, before the app gives up and just closes the entire menu.

**To Reproduce**
1. Have a moderate number (think hundreds) of soundfonts
2. Attempt to change soundfonts.
3. Fail

**Expected behavior**
I'd expect a scrollable menu to appear in less than a full minute, and work. Well, actually, I'd expect a menu where I can actually choose a soundfont, then a patch, instead of an extremely long dropdown with just patches. But I guess that'll happen in due time. I can just use JuicySF for now..._is what I would say if it appeared on the list of VSTs_.

**Screenshots**

Notice how the background is greyed out. Because the whole program is hanging.
![image](https://user-images.githubusercontent.com/13303368/206068085-145920f5-5cab-4e41-9c69-14a63a0e92b5.png)

**Platform information**
 - OS: Windows 10

**Comment by DmitryArefiev** [comment]
Also see #25511

**Comment by DmitryArefiev** [comment]
Can't reproduce the freeze because I don't have a hundreds of SFs. 

But with my **~40 SFs**, I also see a **sluggishness** when hovering over SoundFonts group, and that **makes it difficult to operate sound founts in mixer** (if user has a lot of them):


https://github.com/user-attachments/assets/ce015ede-572f-42b7-903c-c47983291b98

**Comment by DaforLynx** [comment]
> @DaforLynx Can you recheck it in the latest release please?

@DmitryArefiev Yep. Installed latest release from musescore.org, now it hangs even longer - forever, in fact (I've collected a good deal more individual soundfonts since then - totaling somewhere in the thousands).

Why is this not a file explorer window?

**Comment by Eism** [comment]
It looks like the problem in our menu (in calculating size)

<img width="2122" alt="Screenshot 2024-11-21 at 3 18 28 PM" src="https://github.com/user-attachments/assets/d463488c-de8f-4a61-8030-c9bb6fdc0971">

**Comment by DmitryArefiev** [comment]
> Why is this not a file explorer window?

That's a new design

> 
Edit: In fact, nothing seems to work. VSTs don't load a GUI and crash the whole program when attempting to playback. 

@DaforLynx Unfortunately we had a crashes with some VSTs.. Please log a separate bugs for that, if they have not yet been logged (use search: https://github.com/musescore/MuseScore/issues?q=is%3Aissue+is%3Aopen+label%3AVST+%2B+crash )

**Comment by DmitryArefiev** [comment]
@bkunda Since the current issue got worse (after we implemented grouping for SoundFonts in Mixer) and also there is another report #25511, I propose trying to fix it in 4.5 (with P3 for now)
